### PR TITLE
Code cleanup, better coverage, readme improvements

### PR DESCRIPTION
**Pull request by noffle**
The README example could still use work (e.g. demonstrate how we might use it in js-ipfs proper).

### CODE REVIEW

File: test/repo-test.js [hunk 1]
```diff
@@ -14,6 +15,32 @@ const fileAExt = fs.readFileSync(join(__dirname, 'test-repo/blocks/12207028/1220
...
+      }
+      expect(run).to.throw(Error)
+      done()
+    })
+
+    it('bad repo init 2', (done) => {
+      function run () {
+        new Repo('', {})
+      }
+      expect(run).to.throw(Error)
```
Comment (dignifiedquire) [inline test/repo-test.js]: There is a nice way I found for doing this using arrow functions 

```js
expect(
  () => new Repo('', {})
).to.throw(Error)
```

File: src/index.js [hunk 2]
```diff
@@ -51,16 +51,6 @@ function Repo (repoPath, options) {
   this.datastore = stores
                 .datastore
                 .setUp(repoPath, options.stores.datastore, this.locks)
-
-  // TODO: needs https://github.com/ipfs/js-ipfs-repo/issues/6#issuecomment-164650642
-  // this.datastoreLegacy = stores
-  //              .datastore
-  //              .setUp(repoPath, options.stores.datastore, this.locks)
```
Comment (diasdavid) [inline src/index.js]: This is not "unused code" It is a TODO comment

**Comment by diasdavid** [comment]
Seems that linting is failing cause of the `new` without assignment

File: src/index.js (hunk 2 above)
Comment (noffle) [inline src/index.js]: Let's make a tracking issue for the work then -- that's the proper place for it. Issues embedded in code have a very low visibility and discoverability.

**Comment by noffle** [comment]
Managed to trick the linter.

**Comment by diasdavid** [comment]
woot. Thank you @noffle :)

**Pull request merged**